### PR TITLE
GestureState helper

### PR DESCRIPTION
**Pull request by bcardarella**
* failing test to make green

### CODE REVIEW

File: test/live_view_native/swiftui/rules_parser_test.exs [hunk 1]
```diff
@@ -366,6 +366,18 @@ defmodule LiveViewNative.SwiftUI.RulesParserTest do
 
       assert parse(input) == output
     end
+
+    test "gesture" do
+      input = ~s{offset(x: gesture_state(:drag, .translation.width))}
+      output = {:offset, [], [x: {:__gesture_state__, [], [:drag, {:., [], [nil, {:., [], [:translation, :width]}]}]}]}
```
Comment (NduatiK) [inline test/live_view_native/swiftui/rules_parser_test.exs:372]: @bcardarella, can the output be the following (changes marked with `^`s)?

```elixir
output = {:offset, [], [[x: {:__gesture_state__, [], [:drag, {:., [], [nil, {:., [], [:translation, :width]}]}]}]]}
                        ^                                                                                        ^
```

The pattern so far has been to parse a key value list into a list:
```elixir
input = ~s{searchable(change: event("search-event", throttle: 10_000))}
output = {:searchable, [], [[change: {:__event__, [], ["search-event", [throttle: 10_000]]}]]}
      
input = "background(content: :star_red)"
output = {:background, [], [[content: :star_red]]}
```

Comment (carson-katri) [reply test/live_view_native/swiftui/rules_parser_test.exs:372]: Yeah, I would prefer for the client if it followed the same format as a regular modifier would, just with `gesture_state` replaced with `__gesture_state__`

Comment (NduatiK) [reply test/live_view_native/swiftui/rules_parser_test.exs]: Got it, I've pushed a change with the updated output

File: lib/live_view_native/swiftui/rules_parser/post_processors.ex [hunk 2]
```diff
@@ -6,23 +6,21 @@ defmodule LiveViewNative.SwiftUI.RulesParser.PostProcessors do
...
       empty()
       |> PostProcessors.inspect()
       |> combinator
-      |> PostProcessors.inspect()
-
+      |> PostProcessors.inspect(label: "combinator")
     This function is extremely useful for debugging, do not remove
     """
-    def inspect(combinator) do
+    def inspect(combinator, opts \\ []) do
```
[the remark this answers is not in the document]
Comment (NduatiK) [reply lib/live_view_native/swiftui/rules_parser/post_processors.ex:15]: We're planning to merge both parsers eventually, right? Can we do it then?

[the remark this answers is not in the document]
Comment (bcardarella) [reply lib/live_view_native/swiftui/rules_parser/post_processors.ex:15]: yup, that works :+1:

**Comment by bcardarella** [comment]
I'll let @carson-katri sign off on this next week

File: lib/live_view_native/swiftui/rules_parser/modifiers.ex [hunk 3]
```diff
@@ -211,6 +220,11 @@ defmodule LiveViewNative.SwiftUI.RulesParser.Modifiers do
         parsec(:event),
         ~s'an event eg ‘event("search-event", throttle: 10_000)’'
       },
+      {
+        parsec(:gesture_state),
+        nil
+        # ~s'an event eg ‘event("search-event", throttle: 10_000)’'
```
Comment (carson-katri) [inline lib/live_view_native/swiftui/rules_parser/modifiers.ex:226]: Should this line be removed?

Comment (NduatiK) [reply lib/live_view_native/swiftui/rules_parser/modifiers.ex:226]: Yes, removed

**Pull request merged**